.github/workflows/main.yaml

```diff
@@ -290,5 +289,5 @@ jobs:
         run: |
           cargo clippy
           cargo fmt --check
-          cargo run --example=example
-          cargo test --verbose
+          cargo run --example=example --features compile-stan-model
+          cargo test --verbose --features compile-stan-model -- --test-threads=1
```

Comment:
I'm a bit concerned that --test-threads=1 will have some impact on our other tests, but I don't recall entirely what it means. I think @aseyboldt and I discussed this when the rust bindings were initially developed

Reply:
This is the number of threads that are ran in parallel, the reason I added this is because there is a race condition on the compile test because it deletes the .so file before proceding with the compilation and this can interfere with the other tests.

Reply:
as an alternative i've added a "ignore" flag to this new test case (on the rust file), so then `cargo test` won't run it. So, then added a new line to main.yaml to run just this test after the other ones.